engine/src/test/java/io/camunda/zeebe/streamprocessor/ProcessingScheduleServiceTest.java

```diff
@@ -288,7 +288,7 @@ public void shouldScheduleOnFixedRate() {
     scheduleService.runAtFixedRate(Duration.ofMillis(10), mockedTask);
 
     // then
-    verify(mockedTask, TIMEOUT.times(5)).execute(any());
+    verify(mockedTask, TIMEOUT.atLeast(5)).execute(any());
```

Comment:
It doesn't look like this test is testing really what it wants. To make it deterministic, one way would be to use `ControlledActorScheduler`. Then we can write:

```
    scheduleService.runAtFixedRate(Duration.ofSeconds(10), mockedTask);
    actorScheduler.runUntilDone();

    // verify it ran at the first interval
    clock.addTime(Duration.ofSeconds(10));
    actorScheduler.runUntilDone();
    verify(mockedTask, times(1)).execute(any());
 
   // verify it was rescheduled for the next interval
    clock.addTime(Duration.ofSeconds(10));
    actorScheduler.runUntilDone();
    verify(mockedTask, times(2)).execute(any());
```

What do you think?

Reply:
I interpreted the test more as a "see if `runAtFixedRate` works at all" test and not so much as a test that verifies specific timings.

I think it makes sense to test that `runAtFixedRate` runs jobs at a rate that is close to the requested rate but that'd be a separate test IMO.

Reply:
Ok. Then this test is enough. I don't think another test is required. I assume time based scheduling is already tested for actor scheduler.